### DISCUSSION TITLE
How to check if a point (yaw, pitch, zoomLvl) is in the photo?

### DISCUSSION THREAD

**Discussion by yangwk4464** [mistic100/Photo-Sphere-Viewer#1042] [General]
For example, there is a marker in a special position (yaw, pitch, zoomLvl) 

How to calculate if this position is in the current view or not?

Is there any util in Photo-Sphere-Viewer user could leverage?

**Comment by mistic100** [comment]
```js
viewer.dataHelper.sphericalCoordsToViewerCoords({ yaw, pitch });
```
will return the position of a given yaw/pitch point in the viewer element coordinates. it does not check for actual visibility but this can be done by comparing the value to the actual element size

```js
const { x, y } = viewer.dataHelper.sphericalCoordsToViewerCoords({ yaw, pitch });
const { width, height } = viewer.state.size;

const visible = x > 0 && x < width && y > 0 && y < height;
```

**Reply by yangwk4464** [reply]
Thanks for the Reply! This is exactly what I need!

I found small issue if we use sphericalCoordsToViewerCoords directly. 
Because sphericalCoordsToViewerCoords will treat the opposite side the same effect, which means there if we rotate 180 degrees, the "visible" still be true. Therefore we need to check the target spot's yaw/pitch and current position's yaw/pitch as well in the visible calcuation.

Anyway it works very well, Thanks!

**Reply by mistic100** [reply]
Oups you are right. The markers plugin does this additional check:

```js
viewer.dataHelper.sphericalCoordsToVector3({ yaw, pitch }).dot(viewer.state.direction) > 0
```

it checks if the point is in front of the camera and not in the back
